### PR TITLE
Feature/redis

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

implement data storage to save data to redis

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested? 

Tested locally and unit tests

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS-1633

### CODE REVIEW

File: app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs [hunk 1]
```diff
@@ -1,13 +1,63 @@
...
 
 namespace BcGov.Fams3.Redis
 {
     public interface ICacheService
     {
-        Task SaveRequest(SearchRequest searchRequest);
-        Task<Person> GetRequest(Guid searchRequestId);
+        bool SaveRequest(SearchRequest searchRequest);
+        SearchRequest GetRequest(Guid searchRequestId);
+        bool DeleteRequest(Guid searchRequestId);
```
Comment (rantifamilusi) [inline app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: Can we make this methods async

File: app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs [hunk 2]
```diff
@@ -1,13 +1,63 @@
...
+            try
+            {
+                if (searchRequestId == null) return null;
+                string str = searchRequestId.ToString();
+                string searchRequestStr = _database.StringGet(searchRequestId.ToString(), CommandFlags.None);
+                if (searchRequestStr == null) return null;
+                return JsonConvert.DeserializeObject<SearchRequest>(searchRequestStr);
+            }catch(Exception)
+            {
+                return null;
```
Comment (rantifamilusi) [inline app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: I think we should return InvalidOperation Exception

Comment (rantifamilusi) [reply app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: Maybe not InvalidOperation depending on the exception that can be generated. We should throw the right exception.

File: app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs [hunk 3]
```diff
@@ -1,13 +1,63 @@
...
+            try
+            {
+                if (searchRequest == null) return false;
+                else
+                {
+                    return _database.StringSet(searchRequest.SearchRequestId.ToString(), JsonConvert.SerializeObject(searchRequest));
+                }
+            }catch(Exception)
+            {
+                return false;
```
Comment (rantifamilusi) [inline app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: We should throw the right exception. What kind of exception will happen?

File: app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: sure.

File: app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/BcGov.Fams3.Redis/ICacheService.cs]: good suggestion. will do.

**Pull request merged**